aggregator/generic_elem.go

```diff
@@ -209,6 +212,23 @@ func (e *GenericElem) AddUnion(timestamp time.Time, mu unaggregated.MetricUnion)
 	return nil
 }
 
+// AddValue adds a metric value at a given timestamp.
+func (e *GenericElem) AddValue(timestamp time.Time, value float64) error {
+	alignedStart := timestamp.Truncate(e.sp.Resolution().Window).UnixNano()
+	lockedAgg, err := e.findOrCreate(alignedStart, createAggregationOptions{})
```

Comment:
Is it possible that the `alignedStart` time is earlier than the earliest time that the `elem` will allow?

Reply:
nope, that would be filtered out earlier